extensions/kafka-streams/deployment/src/main/java/io/quarkus/kafka/streams/deployment/KafkaStreamsHotReplacementSetup.java

```diff
@@ -25,22 +26,19 @@ public void setupHotDeployment(HotReplacementContext context) {
 
         @Override
         public void run() {
-            if (nextUpdate < System.currentTimeMillis()) {
-                synchronized (this) {
-                    if (nextUpdate < System.currentTimeMillis()) {
-                        executor.execute(() -> {
-                            try {
-                                context.doScan(true);
-                            } catch (RuntimeException e) {
-                                throw e;
-                            } catch (Exception e) {
-                                throw new RuntimeException(e);
-                            }
-                        });
-                        // we update at most once every 2s
-                        nextUpdate = System.currentTimeMillis() + TWO_SECONDS;
+            final long now = System.currentTimeMillis();
+            // The getAndUpdate we will update the nextUpdate field if it's value is more that now then will return the current value.
+            // If the value is more than now, a new task will be send to the executor.
+            if (nextUpdate.getAndUpdate((value) -> value < now ? value : now + TWO_SECONDS) > now) {
```

Comment:
Probably just me, but I find this entire line hard to grasp. Not sure how to improve it, though.

Reply:
It's not just you :).

Reply:
I propose a new version yesterday https://github.com/quarkusio/quarkus/pull/5445#discussion_r346210543